### ISSUE TITLE
[release-1.23] Backports for 2022-11

### ISSUE THREAD

**Issue by brandond** [k3s-io/k3s#6427]
Backport fixes for the following issues to the `release-1.23` branch:
* #6146
* #6421
* #6422
* #6273
* #6304
* #6423
* #2955
* #6377
* #6424
* #6377
* #5184
* #6358
* #5562
* #6401
* #5562
* #6401
* #6425

**Comment by ShylajaDevadiga** [comment]
### Validated using commit id 2ac863c36b1747065949979a7df5a6581cb58f05 from release-1.23 branch
#6146 Allow not stripping binaries in order to attach a debugger
**no testing needed**

#6421 containerd config version 1 is deprecated 

<details>
Validation containerd version is updated to version 2

```
$ sudo cat /var/lib/rancher/k3s/agent/etc/containerd/config.toml

version = 2

[plugins."io.containerd.internal.v1.opt"]
  path = "/var/lib/rancher/k3s/agent/containerd"
[plugins."io.containerd.grpc.v1.cri"]
  stream_server_address = "127.0.0.1"
  stream_server_port = "10010"
  enable_selinux = false
  sandbox_image = "rancher/mirrored-pause:3.6"

[plugins."io.containerd.grpc.v1.cri".containerd]
  snapshotter = "overlayfs"
  disable_snapshot_annotations = true


[plugins."io.containerd.grpc.v1.cri".cni]
  bin_dir = "/var/lib/rancher/k3s/data/586c81e8fc30a443aec62bf87fa8e441e9d78d124f4110d951e2cb37735f344a/bin"
  conf_dir = "/var/lib/rancher/k3s/agent/etc/cni/net.d"


[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
  runtime_type = "io.containerd.runc.v2"

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
  SystemdCgroup = true

$ sudo crictl -r unix:///run/k3s/containerd/containerd.sock ps -a
CONTAINER           IMAGE               CREATED             STATE               NAME                     ATTEMPT             POD ID
0b8491f141034       288889429becf       4 hours ago         Running             traefik                  0                   37d1d6d69c7df
99e61ce99686c       dbd43b6716a08       4 hours ago         Running             lb-tcp-443               0                   dd512d78e2ed1
c718684f1746c       dbd43b6716a08       4 hours ago         Running             lb-tcp-80                0                   dd512d78e2ed1
61028ab12503e       38b3b9ad736af       4 hours ago         Exited              helm                     1                   3f0bf8e56c7dd
0702d06d11101       38b3b9ad736af       4 hours ago         Exited              helm                     0                   f0b5032818e52
d25e4ed616b21       a81c2ec4e946d       4 hours ago         Running             coredns                  0                   96524aa8b3190
0bc86b545deff       e57a417f15d36       4 hours ago         Running             metrics-server           0                   196350f250c64
ca33a5e5ac67a       9621e18c33880       4 hours ago         Running             local-path-provisioner   0                   24f47ae7ea0f6
$
```
Validated snapshotter: stargz works as expected
```
$ sudo cat /var/lib/rancher/k3s/agent/etc/containerd/config.toml

version = 2

[plugins."io.containerd.internal.v1.opt"]
  path = "/var/lib/rancher/k3s/agent/containerd"
[plugins."io.containerd.grpc.v1.cri"]
  stream_server_address = "127.0.0.1"
  stream_server_port = "10010"
  enable_selinux = false
  sandbox_image = "rancher/mirrored-pause:3.6"

[plugins."io.containerd.grpc.v1.cri".containerd]
  snapshotter = "stargz"
  disable_snapshot_annotations = false


[plugins."io.containerd.snapshotter.v1.stargz"]
cri_keychain_image_service_path = "/run/containerd-stargz-grpc/containerd-stargz-grpc.sock"
[plugins."io.containerd.snapshotter.v1.stargz".cri_keychain]
enable_keychain = true

[plugins."io.containerd.grpc.v1.cri".cni]
  bin_dir = "/var/lib/rancher/k3s/data/586c81e8fc30a443aec62bf87fa8e441e9d78d124f4110d951e2cb37735f344a/bin"
  conf_dir = "/var/lib/rancher/k3s/agent/etc/cni/net.d"


[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
  runtime_type = "io.containerd.runc.v2"

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
  SystemdCgroup = true

$ sudo crictl -r unix:///run/k3s/containerd/containerd.sock ps -a
CONTAINER           IMAGE               CREATED             STATE               NAME                     ATTEMPT             POD ID
56c7e06eff588       288889429becf       2 seconds ago       Running             traefik                  0                   37357f98c8e30
f72a2635ff6e1       dbd43b6716a08       3 seconds ago       Running             lb-tcp-443               0                   c86ef81fd52ca
51dac494da47d       dbd43b6716a08       4 seconds ago       Running             lb-tcp-80                0                   c86ef81fd52ca
daa60b9bfb4a4       38b3b9ad736af       8 seconds ago       Exited              helm                     1                   95f587bf6aad2
9ace74c91c6a1       38b3b9ad736af       12 seconds ago      Exited              helm                     0                   e2fb05e980676
af6d598e9d65a       e57a417f15d36       16 seconds ago      Running             metrics-server           0                   83e6470085e41
3cfbe50915d51       a81c2ec4e946d       17 seconds ago      Running             coredns                  0                   601ebd4666481
45eb159eb27a5       9621e18c33880       17 seconds ago      Running             local-path-provisioner   0                   0b65b54f52894
$ 
```
</details>

#6273 Deploy controller stuck at the server could not find the requested resource when manifest includes removed resource
<details>
Reproduced the issue in k3s version v1.23.13+k3s1
</br>
</br>

**Fresh Install**
1. Install k3s
2. Apply the manifest with Addon as discussed in the issue 

Sample manifest
```
kubectl apply -f - <<EOF
apiVersion: k3s.cattle.io/v1
kind: Addon
metadata:
  name: test
  namespace: kube-system
status:
  gvks:
  - Group: invalid
    Kind: BogusKind
    Version: v1
EOF
```
3. Make the manifest file empty
4. Check logs
```
Nov 09 19:09:22 ip-172-31-9-46 k3s[3985]: I1109 19:09:22.683635    3985 event.go:294] "Event occurred" object="kube-system/test" kind="Addon" apiVersion="k3s.cattle.io/v1" type="Warning" reason="ApplyManifestFailed" message="Applying manifest at \"/var/lib/rancher/k3s/server/manifests/test.yaml\" failed: the server could not find the requested resource"
```
Validated fix using commit id 2ac863c36b1747065949979a7df5a6581cb58f05 on release-1.23 branch
Confirmed no warning is seen with message *failed: the server could not find the requested resource* after the manifest file is made empty

**Upgrade**
1. Install k3s version v1.23.13+k3s1
2. Apply the manifest with Addon as discussed in the issue
3. Check logs
4. Upgrade to latest commit on release-1.23 branch 2ac863c36b1747065949979a7df5a6581cb58f05
5. Check logs again to confirm no error message continues to appear

</details>

#6304 Race condition in agent controller and load of rolebinding manifest
Reproduced the issue in k3s version v1.23.13+k3s1 passing --debug --disable=rolebindings during install and delaying applying manifests as mentioned in the issue
<details>

```
Nov 10 00:00:53 ip-172-31-9-46 k3s[37316]: time="2022-11-10T00:00:53Z" level=debug msg="Tunnel authorizer failed to get Kubelet Port: nodes \"ip-172-31-9-46\" is forbidden: User \"system:k3s-controller\" cannot get resource \"nodes\" in API group \"\" at the cluster scope"
```
Validated fix using commit id 2ac863c36b1747065949979a7df5a6581cb58f05 on release-1.23 branch
No such errors are seen while simulating delay in applying manifests
</details>

#6423  Bump dynamiclistener
Waiting for rc
<Details>
Wait for the standard test to be run as part of patch validation
</Details>

#2955 Add CI tests for nf_tables
Validated 
<details>
Validated as part of nightly tests
</details>

#6377 Install test images are Out-of-Date
Validated
<details>
Validated as part of e2e tests
</details>

#6424 Update kine/sqlite version
Validated kine version using go.mod on the release-1.23 branch
<details>
kine
https://github.com/k3s-io/k3s/blob/release-1.23/go.mod#L95
sqlite
https://github.com/k3s-io/k3s/blob/release-1.23/go.mod#L99
</details>

#5184 Enable http3 support for Traefik 
Validation pending
<details>
</details>

#6358 Add multicloud cluster test 
Validated
<details>
Validated as part of e2e tests
</details>

#5562 Create by default traefik ingressClass when creating a cluster
Validation pending
<details>
</details>

#6401 fsGroup does not work with local-path volume
Validation pending
<details>
</details>

#6425 Bump packaged coredns version
Validated coredns version is 1.9.4 and basic dns validation passed
<details>
```
$ kubectl describe pod -n kube-system coredns-5cfbb9f57c-b8qcf |grep -i image
    Image:         rancher/mirrored-coredns-coredns:1.9.4

$ kubectl exec -i -t dnsutils -- nslookup kubernetes.default
Server:		10.43.0.10
Address:	10.43.0.10#53

Name:	kubernetes.default.svc.cluster.local
Address: 10.43.0.1
```
</details>

**Comment by VestigeJ** [comment]
Confirming the following are passing using ARM SLES 15 SP.4
$ k3s -v
```
k3s version v1.23.14-rc1+k3s1 (9995d9a5)
go version go1.17.13
```

Test coverage
- Make sure installation of new charts work
- Make sure helmChartConfig works as expected
- Make sure Deletion of charts works
- Make sure Upgrade of charts work

Installed with no changes (other than the semVer fix)
Installed with manifest files in /var/lib/rancher/k3s/server/manifests/
Remove manifest files post-install
Change manifest files post-install
a. Make them empty
b. Change some of the data in the manifests
c. Replace some of the data (for example, have a Deployment and change it to be a Pod or something)
Add manifest files post-install
Upgrade with no changes
Upgrade with manifest files in /var/lib/rancher/k3s/server/manifests/
a. Change those files after upgrade (same as test 4 above)
<details>
<summary><h4> used dnsutils.yaml for testing sample</h4></summary>

```
apiVersion: v1
kind: Pod
metadata:
  name: dnsutils
  namespace: default
spec:
  securityContext:
     runAsUser: 1000
     runAsGroup: 3000
     fsGroup: 2000
  containers:
  - name: dnsutils
    image: registry.k8s.io/e2e-test-images/jessie-dnsutils:1.3
    command:
      - sleep
      - "3600"
    imagePullPolicy: IfNotPresent
    securityContext:
       allowPrivilegeEscalation: false
  restartPolicy: Always
```
</details>
